### ISSUE TITLE
Boolean used as array not report any error

### ISSUE THREAD

**Issue by integer** [phpstan/phpstan-strict-rules#80]
# Feature request

PHPStan don't report code like this (boolean used as array)
```
$x = FALSE;
var_dump($x['asdf']);
```

### Code snippet that reproduces the problem

https://phpstan.org/r/d76cc038-f93d-4766-a256-2770437d9fee

### Expected output

Error because code uses boolean variable like an array.

**Comment by ondrejmirtes** [comment]
Already implemented: https://phpstan.org/r/5aec307f-0f62-4a61-bd56-65b54d7d1ae4
